shaka-lab-node/macos/log-wrapper.js

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Wrap macOS services for logging purposes.
+//
+// Services based on launchd for macOS can have their output redirected to a
+// log file.  However, this does not rotate logs.  External log rotation
+// mechanisms (such as newsyslog) want to signal the log writer to reopen the
+// log files on rotation, but launchd doesn't support this signal.
+// Therefore, the only way to get service output logging with rotation on macos
+// is to have the service itself write the log files and handle signals.
+//
+// This wrapper allows the service (and its child processes) continue logging
```

Comment:
Nit: It allows the service TO continue logging to etc etc.

Reply:
Fixed